### PR TITLE
Switch to using the livekit Rust SDK

### PR DESCRIPTION
**Pull request by mikayla-maki**
This is part of https://github.com/zed-industries/zed/issues/7015

Release Notes:

- N/A

### CODE REVIEW

File: crates/live_kit_client/src/live_kit_client.rs [hunk 1]
```diff
@@ -1,19 +1,51 @@
...
+    feature = "livekit-rust-sdk"
+))]
+pub mod rust;
+
+#[cfg(all(
+    target_os = "macos",
+    not(any(test, feature = "test-support")),
+    feature = "livekit-rust-sdk"
+))]
+pub use rust::*;
```
Comment (aminya) [inline crates/live_kit_client/src/live_kit_client.rs:35]: I know this is still a work in progress, but this configuration disables the previously working builds on non-MacOS platforms added in #6859

Comment (mikayla-maki) [reply crates/live_kit_client/src/live_kit_client.rs:35]: I appreciate the feedback but this is just to make working with Rust Analyzer easier, I'll make sure to fix everything once we're closer to shipping something like this :)

**Comment by mikayla-maki** [comment]
Switching to working on livekit directly.

**Comment by GaetanLepage** [comment]
Hello !
Is there a new follow up PR for the use of rust livekit ?

**Pull request closed without merging**